### ISSUE TITLE
Add backend method open() and close()

### ISSUE THREAD

**Issue by hagenw** [audeering/audbackend#199]
Copied from https://github.com/audeering/audbackend/issues/197#issuecomment-2017517185

[...] it is very common to open and close connections to a storage drive. So maybe we should actually introduce both functions, e.g.:

```python
backend = audbackend.backend.MyBackend(host, repo)
backend.open()
# do something with the backend
backend.close()
```

or alternatively:

```python
with audbackend.backend.MyBackend(host, repo) as backend:
     # do something with the backend
```

**Comment by hagenw** [comment]
@frankenjoe I will continue working on #197, so I assign this to you (see also https://github.com/audeering/audbackend/pull/200#issuecomment-2024981522)

**Comment by hagenw** [comment]
Solved by #204
